### DISCUSSION TITLE
Mocking Resource Provider in Paparazzi

### DISCUSSION THREAD

**Discussion by savvisingh** [cashapp/paparazzi#1433] [Q&A] [answered]
Hi I m trying to mock the context class inside the Paparazzi so i can mock the strings, My code uses the context.getString inside my compose views and the context used inside paparazzi comes from the sdk, is there a way i can override and mock the context so that i can mock the strings.

**Comment by TWiStErRob** [comment]
Nothing is impossible... have a look at this hackery, and good luck 🙃

https://github.com/TWiStErRob/net.twisterrob.sun/blob/v1.2.2/component/paparazzi/src/main/java/net/twisterrob/sun/test/screenshots/UsableActivityHack.kt

**Reply by savvisingh** [reply, marked as answer]
wow this is some hack, I found more easy way to use 
CompositionLocalProvider(LocalContext provides customContext) to mock the resource in compose views as my views purely compose which i m trying to screenshot

**Reply by TWiStErRob** [reply]
Hah, yes, that's way simpler! Thanks for sharing both question and solution!

**Reply by TWiStErRob** [reply]
By the way, @savvisingh why do you need to mock the strings? Doesn't this work?

```gradle
android.testOptions.unitTests.includeAndroidResources = true
```
